store/tikv/latch/latch.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package latch
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+
+	"github.com/spaolacci/murmur3"
+)
+
+// Latch stores a key's waiting transactions.
+type Latch struct {
+	// A queue by startTs of those waiting transactions.
```

Comment:
Both `acquired` and `waiting` txn is put into `waiting` queue
if `waiting[0] == T`, we can't distinguish T acquired the latch or waiting for the latch, that's not good.
I suggest add a `holder` field, so the case would be much clear.